osu.Server.Queues.ScoreStatisticsProcessor/Commands/Maintenance/MigratePlaylistScoresToSoloScoresCommand.cs

```diff
@@ -0,0 +1,190 @@
+                            });
+
+                            await db.ExecuteAsync("INSERT INTO multiplayer_score_links (user_id, room_id, beatmap_id, playlist_item_id, score_id, created_at, updated_at) VALUES (@userId, @roomId, @beatmapId, @playlistItemId, @scoreId, @createdAt, @updatedAt)", new
+                            {
+                                score.userId,
+                                score.roomId,
+                                score.beatmapId,
+                                score.playlistItemId,
+                                scoreId = insertId,
+                                createdAt = score.started_at,
+                                updatedAt = score.ended_at
+                            });
+
+                            Console.WriteLine();
+                        }
+                    }
+                }
+            }
+
+            Console.WriteLine("Finished.");
+            return 0;
+        }
+
+        private static HitResult getMaxJudgementFor(HitResult hitResult, HitResult max)
+        {
+            switch (hitResult)
+            {
+                case HitResult.Miss:
+                case HitResult.Meh:
+                case HitResult.Ok:
+                case HitResult.Good:
+                case HitResult.Great:
+                case HitResult.Perfect:
+                    return max;
+
+                case HitResult.SmallTickMiss:
+                case HitResult.SmallTickHit:
+                    return HitResult.SmallTickHit;
+
+                case HitResult.LargeTickMiss:
+                case HitResult.LargeTickHit:
+                    return HitResult.LargeTickHit;
+            }
+
+            return HitResult.IgnoreHit;
+        }
+
+        // ReSharper disable InconsistentNaming
+        public class MultiplayerScore
+        {
+            public long id { get; set; }
+            public uint userId { get; set; }
+            public long roomId { get; set; }
+            public long playlistItemId { get; set; }
+            public long beatmapId { get; set; }
+            public string rank { get; set; } = string.Empty;
+            public long? total_score { get; set; }
+            public double? accuracy { get; set; }
+            public double? pp { get; set; }
+            public int max_combo { get; set; }
+            public string mods { get; set; } = string.Empty;
+            public string statistics { get; set; } = string.Empty;
+            public DateTime started_at { get; set; }
+            public DateTime? ended_at { get; set; }
+            public bool passed { get; set; }
+            public DateTime? created_at { get; set; }
+            public DateTime? updated_at { get; set; }
+            public DateTime? deleted_at { get; set; }
+        }
+
+        public class MultiplayerPlaylistItem
+        {
+            public long id { get; set; }
+            public long room_id { get; set; }
```

Comment:
Shouldn't these be

```suggestion
            public ulong id { get; set; }
            public ulong room_id { get; set; }
```

At least that's what I get from

```
MySQL [osu]> describe multiplayer_playlist_items;
+----------------+--------------------+------+-----+---------+----------------+
| Field          | Type               | Null | Key | Default | Extra          |
+----------------+--------------------+------+-----+---------+----------------+
| id             | bigint unsigned    | NO   | PRI | NULL    | auto_increment |
| room_id        | bigint unsigned    | NO   | MUL | NULL    |                |
| owner_id       | int unsigned       | NO   |     | NULL    |                |
| beatmap_id     | mediumint unsigned | NO   |     | NULL    |                |
| ruleset_id     | smallint unsigned  | YES  |     | NULL    |                |
| playlist_order | smallint unsigned  | YES  |     | NULL    |                |
| allowed_mods   | json               | YES  |     | NULL    |                |
| required_mods  | json               | YES  |     | NULL    |                |
| max_attempts   | tinyint unsigned   | YES  |     | NULL    |                |
| created_at     | timestamp          | YES  |     | NULL    |                |
| updated_at     | timestamp          | YES  |     | NULL    |                |
| expired        | tinyint(1)         | NO   |     | 0       |                |
| played_at      | timestamp          | YES  |     | NULL    |                |
+----------------+--------------------+------+-----+---------+----------------+
13 rows in set (0.002 sec)
```

In practice it shouldn't matter for a _very_ long time, but the remaining columns have signedness set correctly, and I'd rather be a stickler for these given how sensitive dapper is to these...